bodo/pandas/series.py

```diff
@@ -585,85 +584,177 @@ def _get_series_python_func_plan(series_proj, empty_data, func_name, args, kwarg
+                    False,  # is_method
+                    (idx,),  # args
+                    {},  # kwargs
+                ),
+                (0,),
+            )
+
+        expr = tuple(create_expr(idx) for idx in range(3))
+
+        # Creates DataFrame with 3 columns
+        df_plan = LazyPlan(
+            "LogicalProjection",
+            empty_data,
+            series_out._plan,
+            expr,
+        )
+
+        return wrap_plan(plan=df_plan)
+
+    return partition
+
+
+def sig_bind(name, accessor_type, *args, **kwargs):
     """
     Binds args and kwargs to method's signature for argument validation.
-    Single exception case is Series.str.wrap() which takes in **kwargs in place of individual
-    keyword arguments. Thus, wrap_signature is manually created, to which the provided arguments are bound.
+    Exception cases, in which methods take *args and **kwargs, are handled separately using sig_map.
+    Signatures are manually created and mapped in sig_map, to which the provided arguments are bound.
     """
+    accessor_names = {"str.": "BodoStringMethods.", "dt.": "BodoDatetimeProperties."}
     msg = ""
     try:
-        if name == "wrap":
-            wrap_params = [
-                inspect.Parameter("width", inspect.Parameter.POSITIONAL_OR_KEYWORD),
-                inspect.Parameter(
-                    "expand_tabs", inspect.Parameter.KEYWORD_ONLY, default=True
-                ),
-                inspect.Parameter(
-                    "replace_whitespace", inspect.Parameter.KEYWORD_ONLY, default=True
-                ),
-                inspect.Parameter(
-                    "drop_whitespace", inspect.Parameter.KEYWORD_ONLY, default=True
-                ),
-                inspect.Parameter(
-                    "break_long_words", inspect.Parameter.KEYWORD_ONLY, default=True
-                ),
-                inspect.Parameter(
-                    "break_on_hyphens", inspect.Parameter.KEYWORD_ONLY, default=True
-                ),
+        if name in sig_map:
+            params = [
+                inspect.Parameter(param[0], param[1])
+                if not param[2]
+                else inspect.Parameter(param[0], param[1], default=param[2][0])
+                for param in sig_map[name]
             ]
-            wrap_signature = inspect.Signature(wrap_params)
-            wrap_signature.bind(*args, **kwargs)
+            signature = inspect.Signature(params)
         else:
-            sample_series = pd.Series(["a"])
-            str_accessor = sample_series.str
-            func = getattr(str_accessor, name)
+            sample_series = (
+                pd.Series(pd.to_datetime(["2023-01-01"]))
+                if accessor_type == "dt."
+                else pd.Series(["a"])
+            )
+            if accessor_type:
+                sample_series = (
+                    sample_series.str if accessor_type == "str." else sample_series.dt
+                )
```

Comment:
This sample series generation is a bit dangerous since it assumes dt/str or Series with string data. More guardrails would be useful.